### ISSUE TITLE
Is model mp_rank_00  correct？

### ISSUE THREAD

**Issue by jiezhangGt** [NVIDIA/Megatron-LM#51]
Following the steps, I downloaded the gpt model（`wget --content-disposition https://api.ngc.nvidia.com/v2/models/nvidia/megatron_lm_345m/versions/v0.0/zip -O megatron_lm_345m_v0.0.zip`） and dictionary & merge files, but the wikitext eval result is not so good, is there something wrong ？

-------------------------------------------------------------------------------------------------------------------------------------------
 validation results on WIKITEXT103 | avg loss: 1.3458E+01 | ppl: 6.9909E+05 | adjusted ppl: 2.7160E+06 | token ratio: 1.1008449901248143 |
-------------------------------------------------------------------------------------------------------------------------------------------

what is more, the gpt2 model I download is release, and my bash script is:
```bash
TASK="WIKITEXT103"

# VALID_DATA=lambada.valid.tokens
VALID_DATA=../wikitext.test.tokens
VOCAB_FILE=checkpoints/gpt2-vocab.json
MERGE_FILE=checkpoints/gpt2-merges.txt
CHECKPOINT_PATH=checkpoints/gpt2_345m

COMMON_TASK_ARGS="--num-layers 24 \
                  --hidden-size 1024 \
                  --num-attention-heads 16 \
                  --seq-length 1024 \
                  --max-position-embeddings 1024 \
                  --fp16 \
                  --vocab-file $VOCAB_FILE"


export CUDA_VISIBLE_DEVICES=7
python3 -u tasks/main.py \
       --task $TASK \
       $COMMON_TASK_ARGS \
       --valid-data $VALID_DATA \
       --strict-lambada \
       --tokenizer-type GPT2BPETokenizer \
       --merge-file $MERGE_FILE \
       --load $CHECKPOINT_PATH \
       --batch-size 8 \
       --checkpoint-activations \
       --log-interval 10 \
       --no-load-optim \
       --no-load-rng
```

The log:
```
using world size: 1 and model-parallel size: 1 
using torch.float16 for parameters ...
-------------------- arguments --------------------
  adlr_autoresume ................. False
  adlr_autoresume_interval ........ 1000
  apply_query_key_layer_scaling ... False
  apply_residual_connection_post_layernorm  False
  attention_dropout ............... 0.1
  attention_softmax_in_fp32 ....... False
  batch_size ...................... 8
  bert_load ....................... None
  bias_dropout_fusion ............. False
  bias_gelu_fusion ................ False
  block_data_path ................. None
  checkpoint_activations .......... True
  checkpoint_num_layers ........... 1
  clip_grad ....................... 1.0
  data_impl ....................... infer
  data_path ....................... None
  DDP_impl ........................ local
  distribute_checkpointed_activations  False
  distributed_backend ............. nccl
  dynamic_loss_scale .............. True
  eod_mask_loss ................... False
  epochs .......................... None
  eval_interval ................... 1000
  eval_iters ...................... 100
  exit_interval ................... None
  faiss_use_gpu ................... False
  finetune ........................ False
  fp16 ............................ True
  fp16_lm_cross_entropy ........... False
  fp32_allreduce .................. False
  hidden_dropout .................. 0.1
  hidden_size ..................... 1024
  hysteresis ...................... 2
  ict_head_size ................... None
  ict_load ........................ None
  indexer_batch_size .............. 128
  indexer_log_interval ............ 1000
  init_method_std ................. 0.02
  keep_last ....................... False
  layernorm_epsilon ............... 1e-05
  lazy_mpu_init ................... None
  load ............................ checkpoints/gpt2_345m
  local_rank ...................... None
  log_interval .................... 10
  loss_scale ...................... None
  loss_scale_window ............... 1000
  lr .............................. None
  lr_decay_iters .................. None
  lr_decay_style .................. linear
  make_vocab_size_divisible_by .... 128
  mask_prob ....................... 0.15
  max_position_embeddings ......... 1024
  merge_file ...................... checkpoints/gpt2-merges.txt
  min_lr .......................... 0.0
  min_scale ....................... 1
  mmap_warmup ..................... False
  model_parallel_size ............. 1
  no_load_optim ................... True
  no_load_rng ..................... True
  no_save_optim ................... False
  no_save_rng ..................... False
  num_attention_heads ............. 16
  num_layers ...................... 24
  num_unique_layers ............... None
  num_workers ..................... 2
  onnx_safe ....................... None
  openai_gelu ..................... False
  overlapping_eval ................ 32
  override_lr_scheduler ........... False
  param_sharing_style ............. grouped
  params_dtype .................... torch.float16
  pretrained_checkpoint ........... None
  query_in_block_prob ............. 0.1
  rank ............................ 0
  report_topk_accuracies .......... []
  reset_attention_mask ............ False
  reset_position_ids .............. False
  save ............................ None
  save_interval ................... None
  scaled_upper_triang_masked_softmax_fusion  False
  seed ............................ 1234
  seq_length ...................... 1024
  short_seq_prob .................. 0.1
  split ........................... 969, 30, 1
  strict_lambada .................. True
  task ............................ WIKITEXT103
  tensorboard_dir ................. None
  titles_data_path ................ None
  tokenizer_type .................. GPT2BPETokenizer
  train_data ...................... None
  train_iters ..................... None
  use_checkpoint_lr_scheduler ..... False
  use_cpu_initialization .......... False
  use_one_sent_docs ............... False
  valid_data ...................... ['../wikitext.test.tokens']
  vocab_file ...................... checkpoints/gpt2-vocab.json
  warmup .......................... 0.01
  weight_decay .................... 0.01
  world_size ...................... 1
---------------- end of arguments ----------------
> building GPT2BPETokenizer tokenizer ...
 > padded vocab (size: 50257) with 47 dummy tokens (new size: 50304)
> initializing torch distributed ...
> initializing model parallel with size 1
> setting random seeds to 1234 ...
> initializing model parallel cuda seeds on global rank 0, model parallel rank 0, and data parallel rank 0 with model parallel seed: 3952 and data parallel seed: 1234
building GPT2 model ...
 > number of parameters on model parallel rank 0: 354871296
global rank 0 is loading checkpoint checkpoints/gpt2_345m/release/mp_rank_00/model_optim_rng.pt
could not find arguments in the checkpoint ...
libibverbs: Warning: no userspace device-specific driver found for /sys/class/infiniband_verbs/uverbs0
  successfully loaded checkpoints/gpt2_345m/release/mp_rank_00/model_optim_rng.pt
 > number of original tokens: 245566, number of detokenized tokens: 270330
> working on iteration: 0
> working on iteration: 10
> working on iteration: 20
> working on iteration: 30
> working on iteration: 40
> working on iteration: 50
> working on iteration: 60
> working on iteration: 70
> working on iteration: 80
> working on iteration: 90
> working on iteration: 100
> working on iteration: 110
> working on iteration: 120
> working on iteration: 130
> working on iteration: 140
> working on iteration: 150
> working on iteration: 160
> working on iteration: 170
> working on iteration: 180
> working on iteration: 190
> working on iteration: 200
> working on iteration: 210
> working on iteration: 220
> working on iteration: 230
> working on iteration: 240
> working on iteration: 250
> working on iteration: 260
> working on iteration: 270
> working on iteration: 280
> working on iteration: 290
> working on iteration: 300
> working on iteration: 310
> working on iteration: 320
> working on iteration: 330
> working on iteration: 340
> working on iteration: 350
> working on iteration: 360
> working on iteration: 370
> working on iteration: 380
> working on iteration: 390
> working on iteration: 400
> working on iteration: 410
> working on iteration: 420
> working on iteration: 430
> working on iteration: 440
> working on iteration: 450
> working on iteration: 460
> working on iteration: 470
> working on iteration: 480
> working on iteration: 490
> working on iteration: 500
> working on iteration: 510
> working on iteration: 520
> working on iteration: 530
> working on iteration: 540
> working on iteration: 550
> working on iteration: 560
> working on iteration: 570
> working on iteration: 580
> working on iteration: 590
> working on iteration: 600
> working on iteration: 610
> working on iteration: 620
> working on iteration: 630
> working on iteration: 640
> working on iteration: 650
> working on iteration: 660
> working on iteration: 670
> working on iteration: 680
> working on iteration: 690
> working on iteration: 700
> working on iteration: 710
> working on iteration: 720
> working on iteration: 730
> working on iteration: 740
> working on iteration: 750
> working on iteration: 760
> working on iteration: 770
> working on iteration: 780
> working on iteration: 790
> working on iteration: 800
> working on iteration: 810
> working on iteration: 820
> working on iteration: 830
> working on iteration: 840
> working on iteration: 850
> working on iteration: 860
> working on iteration: 870
> working on iteration: 880
> working on iteration: 890
> working on iteration: 900
> working on iteration: 910
> working on iteration: 920
> working on iteration: 930
> working on iteration: 940
> working on iteration: 950
> working on iteration: 960
> working on iteration: 970
> working on iteration: 980
> working on iteration: 990
> working on iteration: 1000
> working on iteration: 1010
> working on iteration: 1020
> working on iteration: 1030
> working on iteration: 1040
> working on iteration: 1050
-------------------------------------------------------------------------------------------------------------------------------------------
 validation results on WIKITEXT103 | avg loss: 1.3458E+01 | ppl: 6.9909E+05 | adjusted ppl: 2.7160E+06 | token ratio: 1.1008449901248143 |
-------------------------------------------------------------------------------------------------------------------------------------------
done :-)

```
